packages/contracts-bedrock/deploy/016-L2OutputOracleInit.ts

```diff
@@ -0,0 +1,63 @@
+import { DeployFunction } from 'hardhat-deploy/dist/types'
+import { awaitCondition } from '@eth-optimism/core-utils'
+import '@eth-optimism/hardhat-deploy-config'
+import 'hardhat-deploy'
+
+import {
+  getDeploymentAddress,
+  getContractFromArtifact,
+} from '../src/deploy-utils'
+
+const deployFn: DeployFunction = async (hre) => {
+  const proxyAdmin = await getContractFromArtifact(hre, 'ProxyAdmin')
+  const l2OutputOracle = await getDeploymentAddress(hre, 'L2OutputOracle')
+  const l2OutputOracleProxy = await getDeploymentAddress(
+    hre,
+    'L2OutputOracleProxy'
+  )
+  const l2OutputOracleContract = await getContractFromArtifact(
+    hre,
+    'L2OutputOracle'
+  )
+
+  const startingBlockNumber = hre.deployConfig.l2OutputOracleStartingBlockNumber
+  let startingTimestamp = hre.deployConfig.l2OutputOracleStartingTimestamp
+
+  if (startingTimestamp < 0) {
+    const l1StartingBlock = await hre.ethers.provider.getBlock(
+      hre.deployConfig.l1StartingBlockTag
+    )
+    if (l1StartingBlock === null) {
+      throw new Error(
+        `Cannot fetch block tag ${hre.deployConfig.l1StartingBlockTag}`
+      )
+    }
+    startingTimestamp = l1StartingBlock.timestamp
+  }
+
+  await proxyAdmin.upgradeAndCall(
```

Comment:
The problem here is that there is no key attached to the `proxyAdmin` contract. I believe a way to do it is:

```ts
const [signer] = await hre.ethers.getSigners()
proxyadmin = proxyadmin.attach(signer)
```

Reply:
Got it got it, been working through a bunch of issues